core/dbt/parser/parsing-vs-compilation-vs-runtime.md

```diff
@@ -0,0 +1,151 @@
+# Parsing vs. Compilation vs. Runtime
+
+## Context: Why this doc?
+
+There’s a lot of confusion about what dbt does at parse time vs. compile time / runtime. Even that separation is a relative simplification: parsing includes multiple steps, and while there are some distinctions between "compiling" and "running" a model, the two are **very** closely related.
+
+It's come up many times before, and we expect it will keep coming up! A decent number of bug reports in `dbt-core` are actually rooted in a misunderstanding of when configs are resolved, especially when folks are using pre/post hooks, or configs that alter materialization behavior (`partitions`, `merge_exclude_columns`, etc).
+
+So, here goes.
+
+## What is "parsing"?
+
+**In a sentence:** dbt reads all the files in your project, and constructs an internal representation of the project ("manifest").
+
+To keep it really simple, let’s say this happens in two steps: "Parsing" and "Resolving."
+
+### Parsing
+
+As a user, you write models as SQL + YAML. dbt wants to understand that model as a Python object, defined by an internal data structure. It also wants to know its dependencies and configuration (= its place in the DAG). dbt reads your code **for that one model,** and attempts to construct that object, raising a **validation** error if it can’t.
+
+<details>
+<summary>(Toggle for many more details.)</summary>
+
+- (Because your SQL and YAML live in separate files, this is actually two steps. But for things like `sources`, `exposures`, `metrics`, `tests`, it’s a single pass.)
+- dbt needs to capture and store two vital pieces of information: **dependencies** and **configuration**.
+    - We need to know the shape of the DAG. That includes which models are disabled, in addition to which models will be depending on which other models.
+    - Plus, certain configurations have implications for **node selection**, which supports selecting models using the `tag:` and `config:` methods.
+- Parsing also resolves the configuration for that model, based on configs set in `dbt_project.yml`, and macros like `generate_schema_name`. (These are "special" macros, whose results are saved at parse time!)
+- The way dbt parses models depends on the language that model is written in.
+    - Python models are statically analyzed using the Python AST. [This is pretty fast.](https://www.notion.so/f113894b4cb7412e83b7f9fdb3a24cdb?pvs=21)
```

Comment:
This is a link to an internal Notion doc. We should either remove, or qualify the "pretty fast" by picking a number from the doc

Reply:
Removing